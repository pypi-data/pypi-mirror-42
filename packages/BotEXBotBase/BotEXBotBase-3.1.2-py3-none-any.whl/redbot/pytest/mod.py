import pytest

__all__ = ["mod"]


@pytest.fixture
def mod(config, monkeypatch):
    from redbot.core import Config

    with monkeypatch.context() as m:
        m.setattr(Config, "get_conf", lambda *args, **kwargs: config)
        from redbot.core import modlog

        modlog._register_defaults()
        return modlog
