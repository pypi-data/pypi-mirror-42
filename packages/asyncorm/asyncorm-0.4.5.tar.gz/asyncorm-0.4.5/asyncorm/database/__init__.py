from asyncorm.database.db_manager import PostgresManager
from asyncorm.database.db_cursor import Cursor

__all__ = ["PostgresManager", "Cursor"]
