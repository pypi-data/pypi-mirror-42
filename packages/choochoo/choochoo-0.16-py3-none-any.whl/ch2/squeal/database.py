
from contextlib import contextmanager

from sqlalchemy import create_engine, event
from sqlalchemy.engine import Engine
from sqlalchemy.orm import sessionmaker
from sqlalchemy.sql.functions import count

from . import *
from .support import Base
from ..command.args import DATABASE, NamespaceWithVariables, NO_OP, parser
from ..lib.log import make_log

# mention these so they are "created"
Source,  Interval,
ActivityGroup, ActivityJournal, ActivityTimespan,
Topic, TopicJournal,
StatisticName, StatisticJournal, StatisticJournalInteger, StatisticJournalFloat, StatisticJournalText,
Segment, SegmentJournal
Pipeline
MonitorJournal
Constant, SystemConstant
ActivitySimilarity, ActivityNearby


# https://stackoverflow.com/questions/13712381/how-to-turn-on-pragma-foreign-keys-on-in-sqlalchemy-migration-script-or-conf
@event.listens_for(Engine, "connect")
def fk_pragma_on_connect(dbapi_con, _con_record):
    cursor = dbapi_con.cursor()
    cursor.execute("PRAGMA foreign_keys=ON;")  # https://www.sqlite.org/pragma.html#pragma_foreign_keys
    cursor.execute("PRAGMA temp_store=MEMORY;")  # https://www.sqlite.org/pragma.html#pragma_temp_store
    cursor.execute("PRAGMA threads=4;")  # https://www.sqlite.org/pragma.html#pragma_threads
    cursor.execute("PRAGMA cache_size=-1000000;")  # 1GB  https://www.sqlite.org/pragma.html#pragma_cache_size
    cursor.execute("PRAGMA secure_delete=OFF;")  # https://www.sqlite.org/pragma.html#pragma_secure_delete
    # cursor.execute("PRAGMA journal_mode=WAL;")  # https://www.sqlite.org/wal.html
    cursor.close()


@event.listens_for(Engine, 'close')
def analyxe_pragma_on_close(dbapi_con, _con_record):
    cursor = dbapi_con.cursor()
    cursor.execute("PRAGMA optimize;")  # https://www.sqlite.org/pragma.html#pragma_optimize
    cursor.close()


class Database:

    def __init__(self, args, log):
        self._log = log
        path = args.file(DATABASE)
        self._log.info('Using database at %s' % path)
        self.engine = create_engine('sqlite:///%s' % path, echo=False)
        self.session = sessionmaker(bind=self.engine)
        self.__create_tables()

    def __create_tables(self):
        if self.is_empty(tables=True):
            self._log.info('Creating tables')
            Base.metadata.create_all(self.engine)

    @contextmanager
    def session_context(self):
        session = self.session()
        try:
            yield session
            session.commit()
        except:
            session.rollback()
            raise
        finally:
            session.close()

    def is_empty(self, tables=False):
        if tables:
            # https://stackoverflow.com/questions/33053241/sqlalchemy-if-table-does-not-exist
            return not self.engine.dialect.has_table(self.engine, Source.__tablename__)
        else:
            with self.session_context() as s:
                n_topics = s.query(count(Topic.id)).scalar()
                n_activities = s.query(count(ActivityGroup.id)).scalar()
                n_statistics = s.query(count(StatisticName.id)).scalar()
            return not (n_topics + n_activities + n_statistics)


def add(s, instance):
    s.add(instance)
    return instance


def connect(args):
    '''
    Bootstrap from commandline-like args.
    '''
    p = parser()
    if len(args) == 1:
        args = args[0].split()
    elif args:
        args = list(args)
    else:
        args = []
    args.append(NO_OP)
    ns = NamespaceWithVariables(p.parse_args(args))
    log = make_log(ns)
    db = Database(ns, log)
    return ns, log, db

