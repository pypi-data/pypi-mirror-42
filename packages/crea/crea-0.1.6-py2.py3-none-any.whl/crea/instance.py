# This Python file uses the following encoding: utf-8
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
from builtins import object
import crea as stm


class SharedInstance(object):
    """Singelton for the Crea Instance"""
    instance = None
    config = {}


def shared_crea_instance():
    """ This method will initialize ``SharedInstance.instance`` and return it.
        The purpose of this method is to have offer single default
        crea instance that can be reused by multiple classes.

        .. code-block:: python

            from crea.account import Account
            from crea.instance import shared_crea_instance

            account = Account("test")
            # is equivalent with
            account = Account("test", crea_instance=shared_crea_instance())

    """
    if not SharedInstance.instance:
        clear_cache()
        SharedInstance.instance = stm.Crea(**SharedInstance.config)
    return SharedInstance.instance


def set_shared_crea_instance(crea_instance):
    """ This method allows us to override default crea instance for all users of
        ``SharedInstance.instance``.

        :param Crea crea_instance: Crea instance
    """
    clear_cache()
    SharedInstance.instance = crea_instance


def clear_cache():
    """ Clear Caches
    """
    from .blockchainobject import BlockchainObject
    BlockchainObject.clear_cache()


def set_shared_config(config):
    """ This allows to set a config that will be used when calling
        ``shared_crea_instance`` and allows to define the configuration
        without requiring to actually create an instance
    """
    if not isinstance(config, dict):
        raise AssertionError()
    SharedInstance.config.update(config)
    # if one is already set, delete
    if SharedInstance.instance:
        clear_cache()
        SharedInstance.instance = None
