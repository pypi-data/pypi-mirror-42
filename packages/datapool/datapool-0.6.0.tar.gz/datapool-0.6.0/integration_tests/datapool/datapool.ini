[db]
connection_string = sqlite+pysqlite:////Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz/.simple.db

[backup_landing_zone]
folder = 

[r]
extension = .r
executable = /usr/local/bin/R

[matlab]
extension = .m
executable = /Applications/MATLAB_R2016a.app/bin/matlab

[julia]
extension = .jl
version = 0.5.2
executable = /usr/local/bin/julia

[python]
extension = .py
executable = /Users/uweschmitt/Projects/eawag_datapool/venv/bin/python3

[logging]
config_file = ./logging_config.yaml

[log_receiver]
port = 5559

[worker]
port = 5555
count = 5

[server]
pid_file = /Users/uweschmitt/.datapool/datapool.pid

[landing_zone]
folder = /Users/uweschmitt/Projects/eawag_datapool/src/integration_tests/lz

