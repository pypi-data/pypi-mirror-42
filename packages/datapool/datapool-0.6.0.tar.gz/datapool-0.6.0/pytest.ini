[pytest]
addopts = -rs --full-trace
testpaths = tests
