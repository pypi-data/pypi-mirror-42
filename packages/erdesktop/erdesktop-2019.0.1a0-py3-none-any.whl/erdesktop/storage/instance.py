import os
import peewee

from erdesktop.settings import APP_DB_PATH, APP_DB_FILE


if not os.path.exists(APP_DB_PATH):
	os.makedirs(APP_DB_PATH)

DATABASE_INSTANCE = peewee.SqliteDatabase(APP_DB_FILE)
