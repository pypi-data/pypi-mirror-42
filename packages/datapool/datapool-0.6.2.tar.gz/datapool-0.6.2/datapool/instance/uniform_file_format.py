# encoding: utf-8
from __future__ import print_function, division, absolute_import

import csv
import datetime
import io
import os

from datapool.utils import iter_to_list


def _check_header(header):

    header = list(header)

    if "site" in header:
        header.remove("site")
    else:
        for n in "xyz":
            if n not in header:
                raise ValueError(
                    "header {!r} must have either 'site' or 'x', 'y' and 'z'".format(
                        header
                    )
                )
            header.remove(n)

    header = set(header)
    fix_column_names = set(("timestamp", "parameter", "value"))
    if header != fix_column_names:
        missing = ["'{}'".format(name) for name in fix_column_names - header]
        invalid = ["'{}'".format(name) for name in header - fix_column_names]
        msg = ""
        if missing:
            msg += "missing: {}".format(", ".join(sorted(missing)))
        if invalid:
            msg += "invalid: {}".format(", ".join(sorted(invalid)))
        raise ValueError("invalid header ({})".format(msg))


@iter_to_list
def read_from_file(path):
    if not os.path.exists(path):
        raise ValueError("{} does not exist".format(path))

    with open(path, "r", encoding="ascii") as fh:
        yield from _read_from_fh(fh)


@iter_to_list
def read_from_string(data):
    fh = io.StringIO(data)
    yield from _read_from_fh(fh)


def can_be_converted_to_float(value):
    try:
        value = float(value)
    except ValueError:
        return False
    return True


def parse_timestamp(timestamp):
    return datetime.datetime.strptime(timestamp, "%Y-%m-%d %H:%M:%S")


def _read_from_fh(fh):
    reader = csv.reader(fh, delimiter=";")
    header = [f.strip() for f in next(reader)]
    _check_header(header)
    for i, row in enumerate(reader):
        if len(row) < len(header):
            # i + 1, we skipped header
            raise ValueError("row {} ({!r}) is incomplete".format(i + 1, row))
        if len(row) > len(header):
            # i + 1, we skipped header
            raise ValueError("row {} ({!r}) is to long".format(i + 1, row))

        row = (cell.strip() for cell in row)
        row_dict = dict(zip(header, row))
        yield row_dict


@iter_to_list
def check_rows(row_dicts):

    if not row_dicts:
        return

    for i, row_dict in enumerate(row_dicts):

        value = row_dict["value"]
        if not can_be_converted_to_float(value):
            yield "row {}: value '{}' is not a float".format(i, value)
        else:
            row_dict["value"] = float(value)

        timestamp = row_dict["timestamp"]
        try:
            row_dict["timestamp"] = parse_timestamp(timestamp)
        except ValueError:
            yield "row {}: timstamp '{}' either has wrong format or is invalid".format(
                i, timestamp
            )

        has_xyz = all(name in row_dict and row_dict[name] is not None for name in "xyz")
        not_empty = ("parameter",)
        if not has_xyz:
            not_empty += ("site",)

        for name in not_empty:
            if row_dict[name].strip() == "":
                yield "row {}: field {} is empty".format(i, name)


@iter_to_list
def to_signals(row_dicts):

    for row in row_dicts:
        for f in "xyz":
            if f in row:
                row["coord_" + f] = row[f]
                del row[f]

    from .domain_objects import Signal

    return list(map(Signal, row_dicts))
