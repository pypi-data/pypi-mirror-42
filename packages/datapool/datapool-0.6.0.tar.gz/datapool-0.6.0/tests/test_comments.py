#! /usr/bin/env python
# encoding: utf-8
from __future__ import absolute_import, division, print_function

from datetime import datetime

import pytest
from sqlalchemy.orm import sessionmaker

from datapool.comment_model import add_comment, delete_comment
from datapool.database import dump_db
from datapool.instance.db_objects import (CommentDbo, SignalCommentAssociation,
                                          SignalDbo)
from datapool.instance.domain_objects import Signal
from datapool.instance.signal_model import check_and_commit

from .test_signal_model import engine_with_example_data  # NOQA


@pytest.fixture(scope="function")
def engine_with_signal_and_comment(engine_with_example_data):

    signal = Signal(
        dict(
            signal_id=1000,
            timestamp=datetime(1970, 4, 8),
            site="example_site",
            source="example_source",
            parameter="temperature",
            value=42,
        )
    )
    signals, exceptions = check_and_commit([signal], engine_with_example_data)
    assert not exceptions

    comment_dbo = add_comment(
        engine_with_example_data,
        signal_id=1000,
        text="example comment",
        author="schmittu",
        comment_id=1234,
    )

    assert comment_dbo.comment_id == 1234

    return engine_with_example_data


def test_add_comment(engine_with_signal_and_comment, config, regtest):
    dump_db(config.db, file=regtest)


def test_delete_comment(engine_with_signal_and_comment):
    delete_comment(engine_with_signal_and_comment, 1234)

    session = sessionmaker(bind=engine_with_signal_and_comment)()

    comments = session.query(CommentDbo).all()
    assert len(comments) == 0

    associations = session.query(SignalCommentAssociation).all()
    assert len(associations) == 0


def test_delete_signal(engine_with_signal_and_comment):

    session = sessionmaker(bind=engine_with_signal_and_comment)()
    associations = session.query(SignalCommentAssociation).all()
    assert len(associations) == 1

    signals = session.query(SignalDbo).all()
    assert len(signals) == 1

    session.delete(signals[0])
    session.commit()

    # check deletion cascade:
    associations = session.query(SignalCommentAssociation).all()
    assert not associations
