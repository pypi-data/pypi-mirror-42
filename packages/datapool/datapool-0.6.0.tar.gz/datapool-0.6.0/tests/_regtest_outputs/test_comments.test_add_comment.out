table comment:

   comment_id     text                timestamp               author    
   ----------     ---------------     -------------------     --------  
   1234           example comment     1971-04-08 00:00:00     schmittu  

table parameter:

   parameter_id     name            description     unit  
   ------------     -----------     -----------     ----  
   1                wind speed                      km/h  
   2                temperature                     C     

table parameter_averaging:

   parameter_averaging_id     parameter_id     source_id     integration_length_x     integration_length_y     integration_angle     integration_time  
   ----------------------     ------------     ---------     --------------------     --------------------     -----------------     ----------------  

table picture:

   picture_id     site_id     filename     description     timestamp     data  
   ----------     -------     --------     -----------     ---------     ----  

table project:

   project_id     title     description  
   ----------     -----     -----------  

table quality:

   quality_id     flag     method  
   ----------     ----     ------  

table signal:

   signal_id     value     timestamp               parameter_id     source_id     site_id     coord_x     coord_y     coord_z  
   ---------     -----     -------------------     ------------     ---------     -------     -------     -------     -------  
   1000          42.0      1970-04-08 00:00:00     2                1             1           111         222         None     

table signal_quality:

   signal_quality_id     quality_id     timestamp     author  
   -----------------     ----------     ---------     ------  

table signals_comments_association:

   signal_id     comment_id  
   ---------     ----------  
   1000          1234        

table signals_signal_quality_association:

   signal_id     signal_quality_id  
   ---------     -----------------  

table site:

   site_id     name             description     street     postcode     city     coord_x     coord_y     coord_z  
   -------     ------------     -----------     ------     --------     ----     -------     -------     -------  
   1           example_site                                666          hell     111         222         None     

table source:

   source_id     source_type_id     name               description     serial     manufacturer     manufacturing_date  
   ---------     --------------     --------------     -----------     ------     ------------     ------------------  
   1             1                  example_source     None            000000     None             None                

table source_type:

   source_type_id     name                    description  
   --------------     -------------------     -----------  
   1                  example_source_type     None         

table special_value_definition:

   special_value_definition_id     source_type_id     description     categorical_value     numerical_value  
   ---------------------------     --------------     -----------     -----------------     ---------------  

